dizoo/evogym/envs/evogym_env.py

```diff
@@ -0,0 +1,174 @@
+from typing import Any, Union, List, Optional
+import copy
+import numpy as np
+from easydict import EasyDict
+import gym
+import evogym.envs
+from evogym import WorldObject, sample_robot
+from .viewer import DingEvoViewer
+from evogym.sim import EvoSim
+import os
+from ding.envs import BaseEnv, BaseEnvTimestep, FinalEvalRewardEnv
+from ding.envs.common.common_function import affine_transform
+from ding.torch_utils import to_ndarray, to_list
+from ding.utils import ENV_REGISTRY
+
+
+@ENV_REGISTRY.register('evogym')
+class EvoGymEnv(BaseEnv):
+
+    @classmethod
+    def default_config(cls: type) -> EasyDict:
+        cfg = EasyDict(copy.deepcopy(cls.config))
+        cfg.cfg_type = cls.__name__ + 'Dict'
+        return cfg
+
+    config = dict(
+        env_id='Walker-v0',
+        robot='speed_bot',  # refer to 'world data' for more robots configurations
+        robot_h=5,  # only used for random robots
+        robot_w=5,  # only used for random robots
+        robot_pd=None,  # only used for random robots, probability distributions of randomly generated components)
+        robot_dir=""  # only used for defined robots, path to the robot config, env/world_data/my_bot.json
+    )
+
+    def __init__(self, cfg: dict) -> None:
+        self._cfg = cfg
+        self._init_flag = False
+        self._replay_path = None
+        if 'robot_dir' not in self._cfg.keys():
+            self._cfg = '../'
+
+    def reset(self) -> np.ndarray:
+        if not self._init_flag:
+            self._env = self._make_env()
+            self._env.observation_space.dtype = np.float32  # To unify the format of envs in DI-engine
+            self._observation_space = self._env.observation_space
+            self.num_actuators = self._env.get_actuator_indices('robot').size
+            # by default actions space is double (float64), create a new space with type of type float (float32)
+            self._action_space = gym.spaces.Box(
+                low=0.6, high=1.6, shape=(self.num_actuators, ), dtype=np.float32)
+            self._reward_space = gym.spaces.Box(
+                low=self._env.reward_range[0], high=self._env.reward_range[1], shape=(1, ), dtype=np.float32
+            )
+            self._init_flag = True
+        if hasattr(self, '_seed') and hasattr(self, '_dynamic_seed') and self._dynamic_seed:
+            np_seed = 100 * np.random.randint(1, 1000)
+            self._env.seed(self._seed + np_seed)
+        elif hasattr(self, '_seed'):
+            self._env.seed(self._seed)
+        if self._replay_path is not None:
+            gym.logger.set_level(gym.logger.DEBUG)
+            # use our own 'viewer' to make 'render' compatible with gym
+            self._env.default_viewer = DingEvoViewer(EvoSim(self._env.world))
+            self._env.__class__.render = self._env.default_viewer.render
+            self._env.metadata['render.modes'] = 'rgb_array'  # make render mode compatible with gym
+            self._env = gym.wrappers.Monitor(
```

Comment:
gym.wrappers.Monitor seems to be abandoned in gym>=0.25.
Is the rendering method being tested in gym>=0.25?
If the origin environment does not support gym>=0.25, an "if else" should be applied for different versions.

Reply:
I saw that other envs also used both `Monitor` and `RecordVideo` video class, maybe they should all be updated, for now it can remove Monitor since it is outdated